data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -137,11 +139,43 @@ protected long pos(T record) {
           CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
           deleteSchema.asStruct());
 
-      filteredRecords = Deletes.filter(filteredRecords,
-          record -> projectRow.wrap(asStructLike(record)), deleteSet);
+      Predicate<T> isInDeleteSet = record -> deleteSet.contains(projectRow.wrap(asStructLike(record)));
+      isInDeleteSets.add(isInDeleteSet);
     }
 
-    return filteredRecords;
+    return isInDeleteSets;
+  }
+
+  public CloseableIterable<T> findEqualityDeleteRows(CloseableIterable<T> records) {
```

Comment:
Does this method need to be public? I'm surprised that it is given that `applyEqDeletes` is not.

I also think it would be good to have a better name for it, if it does need to be public. This is really just applying a filter, so I think something like `keepDeletedRows` would be more descriptive.